gui/widgets/list.cpp

```diff
@@ -571,16 +574,40 @@ void ListWidget::drawWidget() {
 				color = _listColors[_listIndex[pos]];
 		}
 
+		Common::Rect r1(_x + r.left, y, _x + r.right, y + fontHeight - 2);
+
+		if (g_gui.useRTL()) {
+			if (_scrollBar->isVisible()) {
+				r1.translate(_scrollBarWidth, 0);
+			}
+
+			if (_numberingMode != kListNumberingOff) {
+				r1.translate(-rtlPad, 0);
+			}
+		}
+
 		if (_selectedItem == pos && _editMode) {
 			buffer = _editString;
 			color = _editColor;
 			adjustOffset();
-			g_gui.theme()->drawText(Common::Rect(_x + r.left, y, _x + r.right, y + fontHeight - 2), buffer, _state,
-			                        Graphics::kTextAlignLeft, inverted, pad, true, ThemeEngine::kFontStyleBold, color);
+			g_gui.theme()->drawText(r1, buffer, _state,
+									alignment, inverted, pad, true, ThemeEngine::kFontStyleBold, color);
```

Comment:
Both of these `g_gui.theme()->drawText()` calls are equivalent, thus, they could be moved out of the `if` statement and only one copy are left.